### PR TITLE
Fix #4301: Remove leading/trailing whitespace from playername

### LINKED ISSUE #4301: leading / trailing whitespace in player name should removed
**OS:** Windows 10
**Version:** 0.0.5
**Commit/Build:** ae05e0f

Player name can't be duplicated in normal ways if player is registed to server(it will became `Player #2`) but this can be bypassed by adding space to end of player's name. this can be confusing since there's no way to distinguish `"Player"` and `"Player "`.

trailing whitespace should removed by client and server so there won't be any confusion.
- [ ] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

### PR DESCRIPTION
**Pull request by Keatzee**
This is my first commit and I'm still learning the codebase, so go easy :smirk: 

This implements #4301 and removes any leading or trailing whitespace from a username. There are two checks: once upon starting the game when the config file is read, another in Network::AddPlayer(). I used String::Trim(), I hope that's the best function to use.

This will only remove leading and trailing whitespace, and not spaces inside of the username.  For example, `_John Peep_` (where '_' is whitespace) should become `John Peep`. 

If another player with the same text but a different number of whitespaces tries to connect afterwards, e.g. `__John Peep__` should become `John Peep #2` as it would if there wasn't whitespace.

Will require network version increase.

### CODE REVIEW

**Comment by Keatzee** [comment]
Still figuring out this Github stuff too...

**Comment by Gymnasiast** [comment]
Never merge other branches into yours. I'll squash the lot for you.

**Comment by Nubbie** [comment]
Welcome, I know too well the trouble I had with Github :p

**Comment by Gymnasiast** [comment]
Thanks for your contribution! I fixed your creative (:P) use of Git.

Some things to watch out for w/r/t Git:
- Never merge other branches into yours. We are the main branch. Your branch will be merged into ours, not the other way round. If your branch gets too far behind, you should rebase instead.
- Do not start your commits with a hash (#). When rebasing, committing and merging interactively, it has a special meaning if it's the first character.

**Comment by Gymnasiast** [comment]
@IntelOrca What do you think, do you merge this for 0.1.0 or 0.1.1? Servers usually use develop, so the latter might be a better idea.

File: src/openrct2/config/Config.cpp [hunk 1]
```diff
@@ -357,6 +357,10 @@ namespace Config
                 }
             }
 
+            // Trim any whitespace before or after the player's name,
+            // to avoid people pretending to be someone else
+            playerName = String::Trim(playerName);
```
Comment (IntelOrca) [inline src/openrct2/config/Config.cpp]: You need to free playerName before replacing it.

**Comment by IntelOrca** [comment]
@Gymnasiast I don't think its too risky to put it in 0.1.0, but it is also not major issue at this time. It also does not prevent empty player names which I believe is an issue still?

**Comment by Gymnasiast** [comment]
@IntelOrca I can't remember if it's still an issue. Regardless, I think this would be better put on the 0.1.1, there is enough stuff to fix for 0.1.0.

**Comment by Keatzee** [comment]
@Gymnasiast Thanks... don't know how I screwed it up as bad as I did :stuck_out_tongue: Git terminology is vague at best.

@IntelOrca I assumed [this code here](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/network/network.cpp#L1480) took care of the empty player name problem, although I'm not sure what it does when the username is empty. There's another check [in Config.cpp](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/config/Config.cpp#L351) that sets it to the logged-in user's name with "Player" as a fallback, but of course that's only called when the game starts.

File: src/openrct2/config/Config.cpp (hunk 1 above)
Comment (Keatzee) [inline src/openrct2/config/Config.cpp]: Is it safe to use `free()` for that? Or is there a better way to do it?

Comment (IntelOrca) [inline src/openrct2/config/Config.cpp]: Yes, you can use `Memory::Free(...)`

**Pull request merged**